pkg/reconciler/common/releases.go

```diff
@@ -230,26 +253,29 @@ func targetManifestPath(version string, instance v1alpha1.KComponent) string {
 			return ""
 		}
 	}
-
-	// Append the spec.additionalManifests
-	addManifests := instance.GetSpec().GetAdditionalManifests()
-	urls = make([]string, 0, len(addManifests))
-	urls = append(urls, manifestPath)
-	for _, manifest := range addManifests {
-		url := strings.ReplaceAll(manifest.Url, VersionVariable, version)
-		urls = append(urls, url)
-	}
-
-	return strings.Join(urls, COMMA)
+	return manifestPath
 }
 
 func targetManifestPathArray(instance v1alpha1.KComponent) []string {
-	if len(instance.GetSpec().GetManifests()) > 0 || len(instance.GetSpec().GetAdditionalManifests()) > 0 {
-		// If either spec.manifests or spec.additionalManifests is not empty, we leverage status.manifests
+	pathArrayPath := ""
+	if len(instance.GetSpec().GetAdditionalManifests()) > 0 {
+		// If spec.additionalManifests is not empty, we leverage status.manifests
 		// to save the complete manifest path.
-		return strings.Split(targetManifestPath(TargetVersion(instance), instance), COMMA)
-	}
+		version := TargetVersion(instance)
+		pathArrayPath = targetManifestPath(version, instance)
+		additionalManifestPath := additionalManifestPath(version, instance)
+		pathArrayPath = strings.Join([]string{pathArrayPath, additionalManifestPath}, COMMA)
 
+	} else if len(instance.GetSpec().GetManifests()) > 0 {
+		// If spec.manifests is not empty, we leverage status.manifests
+		// to save the complete manifest path.
+		pathArrayPath = targetManifestPath(TargetVersion(instance), instance)
+	}
+	if pathArrayPath != "" {
+		return strings.Split(pathArrayPath, COMMA)
```

Comment:
Why do we need to call `strings.Join` above and `strings.Split` here? 
Can `pathArrayPath` be string array?

Reply:
pathArrayPath itself could be a string with multiple paths, separated with comma. additionalManifestPath too.
We connect them together, and split with comma, it will end up with an array of all available paths as each item. NOT JUST pathArrayPath and additionalManifestPath, two items.